qiskit/circuit/quantumcircuit.py

```diff
@@ -274,6 +276,8 @@ def __init__(
             raise TypeError("Only a dictionary or None is accepted for circuit metadata")
         self._metadata = metadata
 
+        super().__init__(self._name, len(self._qubits), len(self._clbits), self._parameters)
```

Comment:
`self._parameters` is always `None` at this point, maybe it would be clearer to explicity pass no params